server/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTextAggregatorFactory.java

```diff
@@ -83,6 +83,10 @@ public SignificantTextAggregatorFactory(String name,
         // Note that if the field is unmapped (its field type is null), we don't fail,
         // and just use the given field name as a placeholder.
         this.fieldType = queryShardContext.fieldMapper(fieldName);
+        if (fieldType != null && fieldType.indexAnalyzer() == null) {
+            throw new IllegalArgumentException("Field [" + fieldType.name() + "] has no index analyzer, but SignificantText " +
```

Comment:
s/index analyzer/analyzer/? Most folks don't declare a different index and search analyzers so they'll probably be confused about this message.

Reply:
Yeah that's fair.  _Technically_ it's the index analyzer that is fetched explicitly, but I could see that causing confusion.  Will change!